File: app/src/main/java/com/snowdango/numac/AppDataEditor/AppDataEditorActivity.java
package com.snowdango.numac.AppDataEditor;

import android.content.Intent;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.view.Display;
import android.view.WindowManager;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import com.snowdango.numac.AppListView.AppListViewActivity;
import com.snowdango.numac.R;

import static android.content.pm.ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED;
import static com.snowdango.numac.NumAcMain.NumAcActivity.builder;
import static com.snowdango.numac.NumAcMain.NumAcActivity.list;
import static com.snowdango.numac.NumAcMain.NumAcActivity.metrics;
import static com.snowdango.numac.NumAcMain.NumAcActivity.windowManager;

public class AppDataEditorActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if(metrics == null){
            windowManager = (WindowManager)getSystemService(WINDOW_SERVICE);
            metrics = new DisplayMetrics();
            Display display = windowManager.getDefaultDisplay();
            display.getMetrics(metrics);
        }

        builder = new AlertDialog.Builder(this);
        setContentView(R.layout.activity_appdata);
        setRequestedOrientation(SCREEN_ORIENTATION_UNSPECIFIED);
    }

    public void onResume() {
        super.onResume();
    }

    @Override
    public void onBackPressed(){
        Intent intent = new Intent(this, AppListViewActivity.class);
        startActivity(intent);
        overridePendingTransition(android.R.anim.slide_in_left, android.R.anim.slide_out_right);
    }

    public int searchAppFormList(String appName){
        int appPosition = -1;
        for(int i = 0; i < list.size(); i++){
            if(list.get(i).getAppName().equals(appName)){
                appPosition = i;
                break;
            }
        }
        return appPosition;
    }
}

File: app/src/main/java/com/snowdango/numac/activites/AppDataEditorActivity.kt
package com.snowdango.numac.activites

import android.content.Context
import android.content.Intent
import android.content.pm.ActivityInfo
import android.os.Bundle
import android.util.DisplayMetrics
import android.view.WindowManager
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.snowdango.numac.R

/*
This class is third listen activity .
pressed BackKey , launch AppListViewActivity.
pressed HomeKey , launch NumAcActivity .
begin AppDataEditorFragment.
 */

class AppDataEditorActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (NumAcActivity.metrics == null) {
            NumAcActivity.windowManager = getSystemService(Context.WINDOW_SERVICE) as WindowManager
            NumAcActivity.metrics = DisplayMetrics()
            val display = NumAcActivity.windowManager!!.defaultDisplay
            display.getMetrics(NumAcActivity.metrics)
        }
        NumAcActivity.builder = AlertDialog.Builder(this)
        setContentView(R.layout.activity_appdata)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED
    }

    public override fun onResume() {
        super.onResume()
    }

    override fun onBackPressed() {
        val intent = Intent(this, AppListViewActivity::class.java)
        startActivity(intent)
        overridePendingTransition(android.R.anim.slide_in_left, android.R.anim.slide_out_right)
    }

    fun searchAppFormList(appName: String): Int {
        var appPosition = -1
        for (i in NumAcActivity.list!!.indices) {
            if (NumAcActivity.list!![i].appName == appName) {
                appPosition = i
                break
            }
        }
        return appPosition
    }
}